dashboard/app/models/pd/application/teacher1819_application.rb

```diff
@@ -426,6 +427,36 @@ def check_idempotency
       Pd::Application::Teacher1819Application.find_by(user: user)
     end
 
+    def workshop
+      return unless regional_partner
+
+      # If this application is associated with a G3 partner who in turn is
+      # associated with a specific teachercon, return the workshop for that
+      # teachercon
+      if regional_partner.group == 3
```

Comment:
Some group 3 partners have local summer workshops, instead of teachercon :(

Probably better to check the form data for [regional_partner_workshop_ids](https://github.com/code-dot-org/code-dot-org/blob/a7df359f3d36cd45b6f512306b564f5f6f027b8d/apps/src/code-studio/pd/application/teacher1819/Section4SummerWorkshop.jsx#L71). You could also check form data for [teachercon](https://github.com/code-dot-org/code-dot-org/blob/a7df359f3d36cd45b6f512306b564f5f6f027b8d/apps/src/code-studio/pd/application/teacher1819/Section4SummerWorkshop.jsx#L72) instead of calling `get_matching_teachercon` directly (note the API that supplies that teachercon text uses the same [call](https://github.com/code-dot-org/code-dot-org/blob/5dde0c0ebbe7a29e2105a456d87094157a55d1a4/dashboard/app/serializers/api/v1/pd/regional_partner_workshops_serializer.rb#L22))

Reply:
Right, but aren't those partners just unrepresented in `get_matching_teachercon`, and so we'll fall through to our base case?

Reply:
Presumably yes. Good point.